syncer/streamer_controller.go

```diff
@@ -266,10 +266,9 @@ func (c *StreamerController) closeBinlogSyncer(logtctx *tcontext.Context, binlog
 // Restart restarts streamer controller
 func (c *StreamerController) Restart(tctx *tcontext.Context, pos mysql.Position) error {
 	c.Lock()
-	defer c.Unlock()
-
 	c.meetError = false
 	c.closed = false
+	c.Unlock()
```

Comment:
Is there any problem between this `c.Unlcok` and `c.Lock` in `ResetReplicationSyncer`?

How about refining the methods into two classes:
- private methods without `Lock`
- public methods with `Lock`

like
```go
func (c *StreamerController) ResetReplicationSyncer (...) {
	c.Lock()
	defer c.Unlock()
	c.resetReplicationSyncer(...)
}

func (c *StreamerController) resetReplicationSyncer (...) {
	...
}
```

Reply:
refine in bd851a0